Se necesita automatizar el despliegue de la aplicación para ejecutar tests

Se necesita diseñar un contenedor para desplegar la aplicación de manera automática para pasar tests unitarios de la APP.
Referente al producto mínimamente viable Interno 3 (https://github.com/mcarmona99/CineTickets/milestone/7). Esta issue es además parte fundamental dentro del PMV.

Se realizará en esta issue la elección de un contenedor base; y a partir de dicho contenedor, instalar paquetes adicionales necesarios, instalación del task runner, módulos de tests, etc. 

Issue no relacionada con ninguna historia de usuario ya que no se avanza con las mismas en este hito, que es puramente de infraestructura.

Volviendo al principal objetivo del anterior comentario,  debemos de instalar los paquetes adicionales necesarios (actualmente ninguno) así como el task runner y dependencias del mismo.

Empezamos instalando `poetry`, que viene a ser nuestro task runner. Si hacemos `RUN pip install poetry`, nos encontramos con el siguiente error:

 debemos de instalar los paquetes adicionales necesarios (actualmente ninguno) así como el task runner y dependencias del mismo.
 
```
...
    unable to execute 'gcc': No such file or directory
    error: command 'gcc' failed with exit status 1
```

Parece ser que nos falta el compilador de gcc. Procedemos a instalarlo.

Tras evitar ese error instalando gcc, tenemos uno nuevo relacionado con la librería limits.h, output:

usr/local/include/python3.6m/Python.h:11:10: fatal error: limits.h: No such file or directory

Solucionamos este error instalando libc-dev. Tras esto, veo más errores. Voy instalando las dependencias necesarias hasta que consigamos instalar poetry.

Finalmente, tenemos el siguiente `Dockerfile`:

```
# Dockerfile para despliegue automático de la aplicación CineTickes. Ejecución de tests
FROM python:3.6-alpine3.13

RUN apk add gcc libc-dev libffi-dev && pip install poetry
```

En total, la imagen supone un tamaño de 225MB.

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,5 +1,12 @@
-# Dockerfile para despliegue automático de la aplicación CineTickes. Ejecución de tests
-FROM python:3.6-alpine3.13
+# Dockerfile para despliegue automático de la aplicación CineTickets. Ejecución de tests
+FROM python:3.8-alpine3.13
 
 # Instalación del gestor de dependencias
 RUN apk add gcc libc-dev libffi-dev && pip install poetry pypyr
+
+COPY . /app
+
+WORKDIR /app
+
+RUN pypyr installdeps
+CMD ["pypyr", "tests"]

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,0 +1,2 @@
+# Dockerfile para despliegue automático de la aplicación CineTickes. Ejeción de tests
+FROM python:3.6-alpine3.13

--- a/Dockerfile
+++ b/Dockerfile
@@ -4,9 +4,11 @@ FROM python:3.8-alpine3.13
 # Instalación del gestor de dependencias
 RUN apk add gcc libc-dev libffi-dev && pip install poetry pypyr
 
+# Copiamos el proyecto a /app
 COPY . /app
 
+# Cambiamos de directorio a /app
 WORKDIR /app
 
-RUN pypyr installdeps
+# Ejecutamos los tests con el task runner
 CMD ["pypyr", "tests"]

--- a/Dockerfile
+++ b/Dockerfile
@@ -2,4 +2,4 @@
 FROM python:3.6-alpine3.13
 
 # Instalación del gestor de dependencias
-RUN apk add gcc libc-dev libffi-dev && pip install poetry
+RUN apk add gcc libc-dev libffi-dev && pip install poetry pypyr

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,2 +1,5 @@
-# Dockerfile para despliegue automático de la aplicación CineTickes. Ejeción de tests
+# Dockerfile para despliegue automático de la aplicación CineTickes. Ejecución de tests
 FROM python:3.6-alpine3.13
+
+# Instalación del gestor de dependencias
+RUN apk add gcc libc-dev libffi-dev && pip install poetry

--- a/tests.yaml
+++ b/tests.yaml
@@ -2,6 +2,12 @@
 # pypyr tests
 steps:
   - name: pypyr.steps.cmd
+    comment: Installs the project dependencies using poetry.
+    in:
+      cmd: pypyr installdeps
+
+  - name: pypyr.steps.cmd
     comment: Run the project tests using pytest.
     in:
+      jump: installdeps
       cmd: poetry run pytest

